localstack/utils/aws/dead_letter_queue.py

```diff
@@ -68,7 +64,51 @@ def _send_to_dead_letter_queue(source_type, source_arn, dlq_arn, event, error):
             raise Exception(msg)
     elif ':sns:' in dlq_arn:
         sns_client = aws_stack.connect_to_service('sns')
-        sns_client.publish(TopicArn=dlq_arn, Message=message, MessageAttributes=message_attrs)
+        for message in messages:
+            sns_client.publish(TopicArn=dlq_arn, Message=message['MessageBody'],
+                MessageAttributes=message['MessageAttributes'])
     else:
         LOG.warning('Unsupported dead letter queue type: %s' % dlq_arn)
     return dlq_arn
+
+
+def _prepare_messages_to_dlq(source_arn, event, error):
+    messages = []
+    custom_attrs = {
+        'RequestID': {'DataType': 'String', 'StringValue': str(uuid.uuid4())},
+        'ErrorCode': {'DataType': 'String', 'StringValue': '200'},
+        'ErrorMessage': {'DataType': 'String', 'StringValue': str(error)}
+    }
+    if ':sqs:' in source_arn:
+        custom_attrs['ErrorMessage']['StringValue'] = str(error.result)
+        for record in event['Records']:
+            msg_attrs = message_attributes_to_upper(record.get('messageAttributes'))
+            message_attrs = {**msg_attrs, **custom_attrs}
+            messages.append({'Id': record['messageId'], 'MessageBody': record['body'],
+                'MessageAttributes': message_attrs})
+    else:
+        if ':sns:' in source_arn:
+            messages.append({'Id': str(uuid.uuid4()), 'MessageBody': json.dumps(event),
+                'MessageAttributes': custom_attrs})
+        elif ':lambda:' in source_arn:
+
+            if event.get('Records') and 'sns' in event['Records'][0]['EventSource']:
+                for record in event['Records']:
+                    message_attrs = {**record['Sns']['MessageAttributes'], **custom_attrs}
+                    messages.append({'Id': record['Sns']['MessageId'], 'MessageBody': record['Sns']['Message'],
```

Comment:
Can we safely assume that these attributes are always present? Perhaps better, to be on the safe side (to avoid `KeyError`):
```
    sns_rec = record.get('Sns', {})
    messages.append({'Id': sns_rec.get('MessageId'), 'MessageBody': sns_rec.get('Message'),
        'MessageAttributes': message_attrs})
```